### PR TITLE
Removing cosim::utility::shared_mutex

### PR DESCRIPTION
**Pull request by ljamt**
- Removing the fiber friendly cosim::utility::shared_mutex
- Also bumping version as this has not been done after 0.8.3 release

### CODE REVIEW

**Review by davidhjp01 — APPROVED** [review]

**Comment by restenb** [comment]
Looks good. With the removal of fibers and swap to `std::mutex`, are there now any changes that are necessary or beneficial to make to `cosim::utility::file_lock` as well?

I don't think any file operations are currently being done on the thread pool.

File: src/cosim/utility/concurrency.cpp [hunk 1]
```diff
@@ -112,7 +49,7 @@ void file_lock::lock()
     // gives us a chance to yield to the other fiber if the operation would
```
[the remark this answers is not in the document]
Comment (ljamt) [reply src/cosim/utility/concurrency.cpp:49]: I don't know what the correct note for `lock()` should be. Maybe we can use `std::lock` now?

There are mentions about fibers in almost every doc string in `concurrenty.cpp` and `concurrency.hpp`. @kyllingstad, shall we just remove all fiber references in the doc (remove all \pre sections) ?

[the remark this answers is not in the document]
Comment (kyllingstad) [reply src/cosim/utility/concurrency.cpp:49]: In most cases, including this one, "fiber" can just be replaced by "thread". But if you want, I can take care of updating the docs in this PR.

**Comment by kyllingstad** [comment]
I've pushed some doc updates now. I also took the liberty of removing some final traces of fibers in a couple of other places.

**Review by kyllingstad — APPROVED** [review]

**Review by markaren — APPROVED** [review]

**Comment by ljamt** [comment]
> I've pushed some doc updates now. I also took the liberty of removing some final traces of fibers in a couple of other places.

Great! Thanks a lot :)

**Pull request merged**